src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -350,6 +360,28 @@ public bool DiagnosticSourceEnabled
             set { }
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether partial flush is enabled
+        /// </summary>
+        public bool PartialFlushEnabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the minimum number of closed spans in a trace before it's partially flushed
+        /// </summary>
+        public int PartialFlushMinSpans
+        {
+            get => _partialFlushMinSpans;
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentException("The value must be strictly greater than 0", nameof(PartialFlushMinSpans));
```

Comment:
Not just greater, **strictly** greater 😄

Reply:
Hmm, this is new. I don't think we throw on invalid settings anywhere else. Part of me things that we should throw when users are setting this from code, but if a user sets env var `DD_TRACE_PARTIAL_FLUSH_MIN_SPANS=-1`, their app may crash during startup (even if `DD_TRACE_PARTIAL_FLUSH_ENABLED=false`).

Should we consider logging this and setting the default value instead? Or changing the logic so it only throw when setting from code but not from other sources?

Reply:
In https://github.com/DataDog/dd-trace-dotnet/pull/1313/commits/a02af7372630a04734e42ac8718265848fa3b8b2, I changed that to not throw if the value is set from the configuration

Reply:
Thanks.